functions/Invoke-DbaLogShippingRecovery.ps1

```diff
@@ -0,0 +1,305 @@
+Invoke-DbaLogShippingRecovery -SqlServer 'server1' -WhatIf
+
+Shows what would happen if the command were executed.
+
+#>
+    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess = $true)]
+    param
+	(
+        [Parameter(Mandatory=$true, Position=1)]
+		[Alias("ServerInstance", "SqlInstance")]
+        [object]$SqlServer,
+        [Parameter(Mandatory=$false, Position=2, ValueFromPipeline=$true)]
+        [string[]]$Database = $null,
+        [Parameter(Mandatory=$false, Position=3)][switch]$NoRecovery,
+        [Parameter(Mandatory=$false, Position=4)][switch]$Silent,
+        [Parameter(Mandatory=$false, Position=5)][System.Management.Automation.PSCredential]$SqlCredential
+	)
+
+    BEGIN
+    {
+        Write-Message -Message "Attempting to connect to Sql Server.." -Level 2 -Silent $Silent
+		$server = Connect-SqlServer -SqlServer $SqlServer -SqlCredential $SqlCredential
+    }
+
+    PROCESS
+    {
+        Write-Message -Message "Started Log Shipping Recovery" -Level 2 -Silent $Silent
+
+        #region Query setup
+        # Query for retrieving the log shipping information
+        $query = "
+----------------------------------------------------------------
+SELECT  lss.primary_server, lss.primary_database, lsd.secondary_database, lss.backup_source_directory,
+        lss.backup_destination_directory, lss.last_copied_file, lss.last_copied_date,
+        lsd.last_restored_file, sj1.name AS 'copyjob', sj2.name AS 'restorejob'
+FROM    msdb.dbo.log_shipping_secondary AS lss
+        INNER JOIN msdb.dbo.log_shipping_secondary_databases AS lsd ON lsd.secondary_id = lss.secondary_id
+		INNER JOIN msdb.dbo.sysjobs AS sj1 ON sj1.job_id = lss.copy_job_id
+		INNER JOIN msdb.dbo.sysjobs AS sj2 ON sj2.job_id = lss.restore_job_id "
+        
+        # Check if they commandlet is eecuted from a pipeline
+		if ($PSCmdlet.MyInvocation.ExpectingInput) 
+		{
+			$db = $_
+            
+            $query += "WHERE lsd.secondary_database = '$db' "
+		}
+		else
+		{
+            # If just one database is set in the database parameter
+			if($Database -ne $null)
+            {
+			    $db = $Database[0]
+
+                $query += "WHERE lsd.secondary_database = '$db'"
+            }
+		}
+        
+        # Add an extra line to easier read the query when in verbose mode
+        $query +="
+----------------------------------------------------------------"
+        #endregion Query setup
+
+        # Retrieve the log shipping information from the secondary instance
+        try{
+            Write-Message -Message "Retrieving log shipping information from the secondary instance" -Level 5 -Silent $Silent
+            $logshipping_details = Invoke-Sqlcmd2 -ServerInstance $SqlServer -Database 'msdb' -Query $query 
+        }
+        catch
+        {
+            Stop-Function -Message "Error retrieving the log shipping details" -Silent $Silent -InnerErrorRecord $_ -Target $SqlServer
+            Write-Error $_.Exception.Message
```

Comment:
You don't need to use Write-Error anymore: Stop-Function does that for you.
Our default is to not show red exceptions in dbatools. There will be an opt-in into red exceptions (that's where _$Silent_ comes in). Feel free to add the exception message to the message text, if you want to pass that to the user.